jte-maven-plugin/src/main/java/gg/jte/maven/GeneratorMojo.java

```diff
@@ -54,6 +54,8 @@
     @Parameter(readonly = true)
     public String packageName = Constants.PACKAGE_NAME_PRECOMPILED;
 
+    @Parameter(defaultValue = "${project.build.directory}/classes")
```

Comment:
I think that should be
`@Parameter(defaultValue = "${project.build.directory}/generated-resources/jte")`

to match the `targetDirectory` above.

Reply:
Ah, this is where I got stuck on how Maven processes resources. For native image to work we want the resources to end up in the same classpath as the classes, and I couldn't see where that would happen. In Gradle this is handled implicitly by the java plugin and processResources task.

Do you know?

Reply:
Oh you're right. It looks like maven doesn't do any resource copying in the phase we're at with the plugin:
https://stackoverflow.com/questions/33166295/how-can-a-plugin-add-its-own-generated-resources

I'd suggest to leave that directory null then (would result in copying the resources next to the classes).

`@Parameter(defaultValue = null)`

If users would want to have that differently, they will need to take care of the copying themselves.